### ISSUE TITLE
Rule request: Avoid Yoda conditions

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1924]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

In Swift, you need a `var` or `let` keyword to do an assignment inside a condition (`if`, `guard`, etc).
Thus, you only get the disadvantages of [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions), but no benefits.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// should trigger
if 42 == foo {}
guard "str" == bar else { return }
while 10 == baz { }

// shouldn't trigger
if foo == 42 {}
guard bar == "str" else { return }
while baz == 10 { }
if foo == bar {}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just the severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

I think this might cause some false positives, so I'd say opt-in. But it could be enabled by default if we can avoid those false positives.

**Comment by jpsim** [comment]
Good rule, but definitely opt-in.

**Comment by jpsim** [comment]
Done in #1965
